index.js

```diff
@@ -12,18 +25,19 @@ class Stok {
 
   // Create a Hapi server with some custom overrides
   createServer() {
-    this.server = new Hapi.Server();
+    this.serverProxy = new HapiProxy();
+    this.server = this.serverProxy.getServer();
+
+    this.registerModule({
```

Comment:
The Hapi Server is registered as a module so that it's treated the same as everything else during shutdown.